DataDog/datadog-cdk-constructs
# feat: Support Go

### What does this PR do?

Support Golang. 

Some details:
- creates a git tag `v2-1.14.0` to make this commit found easily when we later send it to Go package library
- makes gitignore un-ignore `/dist/go`, where the Go package is outputted, so the Go package code can be tracked by Git and uploaded. Currently the entire `/dist` folder is ignored.
- adds `/dist/go` to `.gitattributes` so that these generated files will be collapsed on GitHub, to avoid making code review difficult.



### Motivation
From [Feature Request: Datadog CDK Construct for Go](https://datadoghq.atlassian.net/browse/FRSLES-665)



### Testing Guidelines
I don't know a way to test it at this stage.
My planned next steps are:
1. After this PR is merged, visit https://pkg.go.dev/github.com/DataDog/datadog-cdk-constructs@v2-1.14.0 to request Go package library to include this repo
2. Install this package in a Go project and test it.
3. Update `README.md`
4. Write a release note for v2-1.14.0

Please let me know if there's a good way to test it at this state, or there are other next steps I should take.



### Additional Notes



### Types of Changes


### Check all that apply
